src/app/component/general/eco-news/create-news/create-news.component.ts

```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
 })
 export class CreateNewsComponent implements OnInit {
 
-  link: string = preparedImageForCreateEcoNews;
+  constructor(private router: Router, 
+              private fb: FormBuilder, 
+              private createEcoNewsService: CreateEcoNewsService) {}
 
-  constructor() { }
+  public createNewsForm = this.fb.group({
+    title: ['', [Validators.maxLength(170), Validators.required]],
+    source: [''],
+    content: ['', [Validators.minLength(20), Validators.required]],
+    tags: this.fb.array([])
+  });
+
+  public filters: Array<any> = [
+    {name: 'News', isActive: false},
+    {name: 'Events', isActive: false}, 
+    {name: 'Courses', isActive: false}, 
+    {name: 'Initiatives', isActive: false}, 
+    {name: 'Ads', isActive: false}
+  ];
+
+  public activeLanguage: string = 'en';
+  public link: string;
+  private preparedLink: string = preparedImageForCreateEcoNews;
+  private date: Date = new Date();
+
+  public languages: Array<any> = [
+    {name: 'Ukrainian', lang: 'ua'},
+    {name: 'English', lang: 'en'},
+    {name: 'Russian', lang: 'ru'},
+  ];
 
   ngOnInit() {
+    this.link = this.preparedLink;
+    this.onChanges();
+  }
+
+  private navigateByUrl(): void {
+    this.router.navigateByUrl('/news');
+  }
+
+  public onChanges(): void {
+    this.createNewsForm.get('source').valueChanges.subscribe(source => {
+      if(!source) {
+        this.link = this.preparedLink;
+      } else {
+        this.link = source;
+      }
+    });
   }
 
-}
+  public onSubmit(): void {
+    console.log(this.createNewsForm.value);
+  }
+
+  private addFilters(filter: any): void { 
+    if(!filter.isActive) {
+      filter.isActive = true;
+      this.createNewsForm.value.tags.push(filter.name);
+    } else {
+      filter.isActive = false;
+      this.createNewsForm.value.tags.forEach((item, index) => {
+        if (item === filter.name) {
+          this.createNewsForm.value.tags.splice(index, 1);
+        }
+      })
+    }
+  }
+
+  public changeLanguage(language: any): void {
+    let formData = {
+      text: this.createNewsForm.value['content'],
+      title: this.createNewsForm.value['title']
+    }  
+
+    this.createEcoNewsService.setTranslationByLang(this.activeLanguage, formData);
+
+    this.activeLanguage = language.lang;
+    let translationData = this.createEcoNewsService.getTranslationByLang(this.activeLanguage);
```

Comment:
Extract the code from line 90 to 94 into a separate method

Reply:
fixed